ethereum/contracts/Campaign.sol

```diff
@@ -29,21 +30,21 @@ contract Campaign {
     function contribute() public payable {
         require(msg.value > minimumContribution);
 
-        approvers[msg.sender] = true;
-        approversCount++;
+        if (approvers[msg.sender] != true) {
+          approvers[msg.sender] = true;
+          approversCount++;
+        }
     }
 
-    function createRequest(string memory description, uint value, address recipient)
-        public restricted {
-        Request memory newRequest = Request({
-           description: description,
-           value: value,
-           recipient: recipient,
-           complete: false,
-           approvalCount: 0
-        });
-
-        requests.push(newRequest);
+    function createRequest(string memory description, uint value, address payable recipient)
+        public restricted returns (uint requestID){
+          requestID = requestsCount++;
+          Request storage newRequest = requests[requestID];
+          newRequest.description = description;
+          newRequest.value = value;
+          newRequest.recipient = recipient;
+          newRequest.complete = false;
+          newRequest.approvalCount = 0;
```

Comment:
This was the biggest thing that broke when the update happened. I'm not totally sure what happened, but I was able to find this type of set up in the Solidity Docs and it is working now.